src/Google.Pubsub.V1/SimplePubsub.cs

```diff
@@ -0,0 +1,215 @@
+                        return new SimpleTopic(this, topicId);
+                    }
+                    catch (RpcException creationException) when (i < RetryCount - 1 && creationException.Status.StatusCode == StatusCode.AlreadyExists)
+                    {
+                        // Probably a race condition... someone else tried to create it just when we did.
+                        // Go back through the loop and check we can get it this time, failing after this happens 3 times.
+                    }
+                }
+            }
+            throw new InvalidOperationException("Bug in SimplePubsub: escaped the retry loop");
+        }
+
+        public async Task<SimpleTopic> GetOrCreateTopicAsync(string topicId)
+        {
+            string topicName = GetTopicName(ProjectId, topicId);
+            const int RetryCount = 3;
+            for (int i = 0; i < RetryCount; i++)
+            {
+                try
+                {
+                    // Just check existence...
+                    await Publisher.GetTopicAsync(topicName);
+                    return new SimpleTopic(this, topicId);
+                }
+                catch (RpcException e) when (e.Status.StatusCode == StatusCode.NotFound)
+                {
+                    try
+                    {
+                        await Publisher.CreateTopicAsync(topicName);
+                        return new SimpleTopic(this, topicId);
+                    }
+                    catch (RpcException creationException) when (i < RetryCount - 1 && creationException.Status.StatusCode == StatusCode.AlreadyExists)
+                    {
+                        // Probably a race condition... someone else tried to create it just when we did.
+                        // Go back through the loop and check we can get it this time, failing after this happens 3 times.
+                    }
+                }
+            }
+            throw new InvalidOperationException("Bug in SimplePubsub: escaped the retry loop");
+        }
+
+        public SimpleSubscription GetSubscription(string subscriptionId)
+        {
+            var subscription = Subscriber.GetSubscription(GetSubscriptionName(ProjectId, subscriptionId));
+            return new SimpleSubscription(this,
+                SubscriptionTemplate.ParseName(subscription.Name)[1],
+                TopicTemplate.ParseName(subscription.Topic)[1]);
+        }
+
+        public async Task<SimpleSubscription> GetSubscriptionAsync(string subscriptionId)
+        {
+            var subscription = await Subscriber.GetSubscriptionAsync(SubscriptionTemplate.Expand(ProjectId, subscriptionId));
+            return new SimpleSubscription(this,
+                SubscriptionTemplate.ParseName(subscription.Name)[1],
+                TopicTemplate.ParseName(subscription.Topic)[1]);
+        }
+
+        public SimpleSubscription GetOrCreateSubscription(string subscriptionId, string topicId, int ackDeadlineSeconds = 60)
```

Comment:
I'd much rather see a TimeSpan here, rather than an int for deadline, which also conforms to the rest of our API.
Using a nullable TimeSpan to allow a default (the value of which should be documented)